### ISSUE TITLE
DBN class does not support no pretraining

### ISSUE THREAD

**Issue by mihaelacr** [mihaelacr/pydeeplearn#7]
The current implementation of DBNs does not support no pretraining (ie. a usual neural network). That feature should be enabled, and then max norm regularization for weight matrices implemented for that case (does not make sense to implement for the pretraining case).

This change might require API changes as well as renaming classes.

**Comment by snurkabill** [comment]
Hi, how are you going to implement this? I may try to do that if you don't have time.

**Comment by mihaelacr** [comment]
Just by changing the current API and allowing no training. Also note that some features related to NNs do not work for DBNs but work in general (eg: batch normalization, L2- weight norm re scaling). 

I plan on working for pydeeplearn tomorrow, this is one of the things I would like to look at.

**Comment by mihaelacr** [comment]
Please see this branch for a DBN version which does not require pretraining. You just need to specify the pre training epochs to be 0 and it will work. 

https://github.com/mihaelacr/pydeeplearn/tree/no-pretrain

Note that I will soon merge this into master and only then I will start adding the features mentioned above.

I will close this issue when the no-pretrain branch will be merged with master.

**Comment by mihaelacr** [comment]
Fixed.
